### PR TITLE
Add Test pipelines

### PR DESCRIPTION
**Pull request by vaikas**
Example test file that I've been playing with

https://gist.github.com/vaikas/b2cf4d96b7601bdccf39a2b34bcc0dc2

### Functional Changes

- [ ] This change can build all of Wolfi without errors (describe results in notes)

Notes:

### SCA Changes

- [ ] Examining several representative APKs show no regression / the desired effect (details in notes)

Notes:

### Linter

- [ ] The new check is clean across Wolfi
- [ ] The new check is opt-in or a warning

Notes:

### CODE REVIEW

**Comment by vaikas** [comment]
ncurses test failure is unrelated, https://github.com/wolfi-dev/os/pull/9128

**Comment by vaikas** [comment]
I rejiggered the commits to more bite size chunks for easier review.

File: docs/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# Testing
+
+Melange provides an ability to test packages with a `test` command. Tests are
+implemented using the same `pipeline`, and `subpackages` centric ways to define
+tests, so it should be very familiar continuation from `build` to `test`.
```
Comment (jspeed-meyers) [inline docs/TESTING.md:5]:
```suggestion
tests, so it should be a very familiar continuation from `build` to `test`.
```

**Review by kaniini — APPROVED** [review]

**Comment by kaniini** [comment]
This looks good to me!

**Pull request merged**